client/src/Dashboard/Dashboard.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import '../index.css';
+import { accountRepository } from '@/_services';
+import { authenticationRepository } from '@/_services';
+import config from 'config';
+const signalR = require('@aspnet/signalr')
+
+class Dashboard extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            accountId: null,
+            currentPlan: null,
+            numberOfUsers: null,
+            maximumNumberOfUsersAllowed: null,
+            pricePerMonth: null,
+            hub: null
+        };
+
+        this.logout = this.logout.bind(this);
+        this.upgradePlan = this.upgradePlan.bind(this);
+    }
+
+    componentDidMount() {
+        const hub = this.configureUserHub();
+        this.start(hub);
+
+        this.createAndGetAccount();
+    }
+
+    createAndGetAccount() {
+        accountRepository.create()
+            .then(id => accountRepository.get(id))
+            .then(createdAccount => {
```

Comment:
I don't like the pattern of "creating" account, then getting it.

* There is no "Creation" of any record which takes place. We simply want to retrieve an existing record (the account) using our credentials.
* Even if a `create` call is needed, there is no reason why it should be followed by a `get`. The first call can just return the resource it just created.

Anyway, I think the core problem is that you don't pass any state from the Login call to this page. The login call could just return the account ID which you then use here to retrieve the record. Another option would be to simply call the `get` endpoint without an ID and let the backend retrieve it based on your authentication.

Reply:
Great point, making changes.